python/lsst/analysis/tools/analysisMetrics/apDiaSourceMetrics.py

```diff
@@ -0,0 +1,78 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = (
+    "NumDiaSourcesMetric",
+    "NumDipolesMetric",
+)
+
+from ..actions.scalar import CountAction
+from ..actions.vector import DownselectVector, FlagSelector
+from ..interfaces import AnalysisMetric
+
+
+class NumDiaSourcesMetric(AnalysisMetric):
+    """Calculate the number of DIA Sources that do not have known
+    bad/quality flags set to true.
+    """
+
+    def setDefaults(self):
+        super().setDefaults()
+
+        self.process.buildActions.flagSelector = FlagSelector()
```

Comment:
Since you are not doing any differential filtering (i.e. making two different filter vectors like one for stars and one for galaxies) I would move this flag selector to the prep stage. 

1. Normally quality flags are filtered there anyway as they would apply to all objects
2. It will make interactive use of your Analysis tool easier. The result of prep stage will directly be all the columns you care about and if necessary you can poke around at it for anything weird prior to sending it to the process stage to have computations done on it.

This would look something like
```suggestion
self.prep.selectors.flagSelector = FlagSelector()
self.prep.selectors.flagSelector.selectWhenFalse = [
            "base_PixelFlags_flag_bad",
            "base_PixelFlags_flag_suspect",
            "base_PixelFlags_flag_saturatedCenter",
            "base_PixelFlags_flag_interpolated",
            "base_PixelFlags_flag_interpolatedCenter",
            "base_PixelFlags_flag_edge",
        ]
self.process.calculateActions.numDiaSources = CountAction(vectorKey="diaSourceId")
self.produce.units = {"numDiaSources": "ct"}
```